File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/idea/wizard/ZeppelinProjectTemplatesFactory.scala
package org.intellij.plugin.zeppelin.idea.wizard


import com.intellij.icons.AllIcons.Icons
import com.intellij.ide.util.projectWizard.WizardContext
import com.intellij.platform.{ProjectTemplate, ProjectTemplatesFactory}
import javax.swing.Icon

class ZeppelinProjectTemplatesFactory extends ProjectTemplatesFactory {
  def createTemplates(group: String,
                      context: WizardContext): Array[ProjectTemplate] = {
    Array(new ZeppelinProjectTemplate())
  }

  override def getGroupIcon(group: String): Icon = Icons.Ide.NextStep

  def getGroups: Array[String] = Array(ZeppelinProjectTemplatesFactory.Group)
}

object ZeppelinProjectTemplatesFactory {
  val Group = "Remote Execute"
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/idea/wizard/ZeppelinProjectTemplatesFactory.kt
package org.intellij.plugin.zeppelin.idea.wizard

import com.intellij.icons.AllIcons.Icons
import com.intellij.ide.util.projectWizard.WizardContext
import com.intellij.platform.ProjectTemplate
import com.intellij.platform.ProjectTemplatesFactory
import javax.swing.Icon

class ZeppelinProjectTemplatesFactory : ProjectTemplatesFactory() {
    override fun createTemplates(group: String?, context: WizardContext?): Array<ProjectTemplate> {
        return arrayOf(ZeppelinProjectTemplate())
    }

    override fun getGroupIcon(group: String): Icon = Icons.Ide.NextStep

    override fun getGroups(): Array<String> = arrayOf(ZeppelinProjectTemplatesFactory.GROUP)

    companion object {
        const val GROUP: String = "Remote Execute"
    }
}